csharp/src/Ice/ConnectionI.cs

```diff
@@ -1655,31 +1643,42 @@ private void setState(int state)
                 case StateActive:
                 {
                     //
-                    // Can only switch from holding or not validated to
-                    // active.
+                    // Can only switch to active from holding or not validated.
                     //
                     if (_state != StateHolding && _state != StateNotValidated)
                     {
                         return;
                     }
+
+                    if (_maxDispatches > 0 && _dispatchCount >= _maxDispatches)
+                    {
+                        // Don't resume reads if we're at or over the _maxDispatches limit.
+                        return;
+                    }
+
                     _threadPool.register(this, SocketOperation.Read);
+                    _idleTimeoutTransceiver?.enableIdleCheck();
```

Comment:
We now enable/disable the idle check when we switch to StateActive resp. StateHolding.